sdk/Trace/Sampler/TraceIdRatioBasedSampler.php

```diff
@@ -29,6 +30,9 @@ class TraceIdRatioBasedSampler implements Sampler
      */
     public function __construct(float $probability)
     {
+        if ($probability < 0.0 || $probability > 1.0) {
+            throw new InvalidArgumentException('probability should be be between 0.0 and 1.0.');
```

Comment:
You may want to add tests around this to validate the cases where probability < 0.0 and probability > 1.0.

Reply:
I added the required tests.